presto-parquet/src/main/java/io/prestosql/parquet/reader/MetadataReader.java

```diff
@@ -293,55 +313,58 @@ private static PrimitiveTypeName getTypeName(Type type)
         }
     }
 
-    private static OriginalType getOriginalType(ConvertedType type)
+    // Copied from org.apache.parquet.format.converter.ParquetMetadataConverter#getLogicalTypeAnnotation(org.apache.parquet.format.ConvertedType, org.apache.parquet.format.SchemaElement)
```

Comment:
We could add a shim to call this in the shaded Hive build, like this: https://github.com/prestosql/presto-hive-apache/blob/master/src/main/java/org/apache/parquet/io/ColumnIOUtil.java

Reply:
https://github.com/prestosql/presto-hive-apache/pull/8